### PR TITLE
Feature/create supplier profile

### LINKED ISSUE #34: page: create supplier profile
job story: when i want to setup my the supplier for my first order, i can create / update a group profile.

screenshot_20170816_145006

will be a similar implementation to https://github.com/root-systems/cobuy/pull/120, possibly done in parallel with https://github.com/root-systems/cobuy/issues/36.

### PR DESCRIPTION
**Pull request by iainkirkpatrick**
closes #34
WIP

adds the Task of creating a Supplier Profile as part of the pre-reqs of signing up

### CODE REVIEW

**Comment by Julietbc** [comment]
end of friday -- next step ----> create relationship record linking group and supplier in relationship table after group + supplier records have been created

File: db/migrations/20170818142530_add-agent-type-ids-to-orders.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+exports.up = function(knex, Promise) {
+  return knex.schema.table('orders', function(table){
+    table.renameColumn('agentId', 'consumerAgentId')
```
Comment (ahdinosaur) [inline db/migrations/20170818142530_add-agent-type-ids-to-orders.js:3]: nice

File: tasks/components/SetupSupplierTask.js [hunk 2]
```diff
@@ -1,6 +1,43 @@
-import React from 'react'
+import h from 'react-hyperscript'
+import { isNil, merge, isEmpty } from 'ramda'
+
+import TaskStepper from './TaskStepper'
+import Profile from '../../agents/components/Profile'
+import MemberInvites from '../../agents/components/MemberInvites'
+
+const rolesToRelationships = (roles = {}) => {
```
Comment (ahdinosaur) [inline tasks/components/SetupSupplierTask.js]: is this used here?

File: tasks/containers/SetupSupplierTask.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        })
+        queries.push({
+          service: 'profiles',
+          params: {
+            query: {
+              agentId: contextAgentId
+            }
+          }
+        })
+        queries.push({
```
Comment (ahdinosaur) [inline tasks/containers/SetupSupplierTask.js]: do we need to query for relationships here?

File: tasks/containers/SetupSupplierTask.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            queries.push({
+              service: 'credentials',
+              params: {
+                query: {
+                  agentId
+                }
+              }
+            })
+          })
+          queryEachMember(members)
```
Comment (ahdinosaur) [inline tasks/containers/SetupSupplierTask.js]: do we need to query for this here?

File: tasks/containers/SetupSupplierTask.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      const { taskPlan } = props.ownProps
+
+      // wait for task plan before re-query
+      if (isNil(taskPlan)) return false
+
+      // re-query when we haven't gotten back contextAgent or taskWork
+      const contextAgent = getContextAgentFromTaskPlan(taskPlan)
+      if (isNil(contextAgent)) return true
+      if (anyMembersAreNil(contextAgent)) {
+        return true
```
Comment (ahdinosaur) [inline tasks/containers/SetupSupplierTask.js]: we might not need this anymore.

File: tasks/services/plans.js [hunk 6]
```diff
@@ -42,11 +42,12 @@ function createChildTaskPlans (hook) {
   const childTaskRecipes = taskRecipe.childTaskRecipes
   return Promise.all(
     childTaskRecipes.map((childTaskRecipe) => {
+      var params = hook.params[childTaskRecipe.id]
       return taskPlans.create({
         parentTaskPlanId: hook.result.id,
         assigneeId: hook.data.assigneeId,
         taskRecipeId: childTaskRecipe.id,
-        params: hook.data.params
+        params: JSON.stringify(params)
```
Comment (ahdinosaur) [inline tasks/services/plans.js]: this should already be done as a before hook on the service, `encodeParams` in https://github.com/root-systems/cobuy/blob/master/tasks/services/plans.js#L21-L32.

**Review by ahdinosaur — COMMENTED** [review]
whoop whoop! 😸 

left some comments

**Comment by iainkirkpatrick** [comment]
waiting on [this PR](https://github.com/root-systems/dogstack-agents/pull/8) before merging

**Comment by ahdinosaur** [comment]
this breaks the "Invite Your Members" page, but we can fix that in #148

**Comment by ahdinosaur** [comment]
in the words of some old friends:

> good enough for government work

good-enough-for-govt-work

**Pull request merged**